pkg/sql/parser/sql.y

```diff
@@ -2222,6 +2222,17 @@ alter_table_cmd:
       ValidationBehavior: $3.validationBehavior(),
     }
   }
+  // ALTER TABLE <name> ADD CONSTRAINT IF NOT EXISTS ...
+| ADD CONSTRAINT IF NOT EXISTS constraint_name constraint_elem opt_validate_behavior
```

Comment:
What if instead of another parser case we added an opt_if_not_exist to the existing rule?

Reply:
last time i tried something like that i got shift reduce'd to oblivion, but i didnt figure out why. you can try but i think that's why even in pg IF EXISTS / NOT EXISTS are always a separate clause.

Reply:
The presence of IF NOT EXISTS makes the constraint_name mandatory instead of optional. There were several ways to deal with this awkwardness and because I don't have taste in these matters I picked the one which made for the smallest diff.